### PR TITLE
policy/api: validate EndpointSelectors at import time

### PR DESCRIPTION
**Pull request by ianvernon**
Before, EndpointSelectors were not validated before being imported into
the policy repository. This meant that users could provide malformed
policy and not be aware that they were doing so, and that Cilium would
accept the policy. Now, as part of the sanitization of rules, validate
the EndpointSelectors in all parts of the rule.

Fixes: #5336

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5709)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

File: pkg/policy/api/rule_validation_test.go [hunk 1]
```diff
@@ -15,7 +15,9 @@
 package api
 
 import (
+	"github.com/cilium/cilium/pkg/labels"
 	. "gopkg.in/check.v1"
```
Comment (rlenglet) [inline pkg/policy/api/rule_validation_test.go:20]: Insert a blank line.

**Review by rlenglet — APPROVED** [review]

**Comment by ianvernon** [comment]
Moving back to wip, I want to see if I can utilize https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/validation/validation.go#L26

File: pkg/policy/api/rule_validation_test.go (hunk 1 above)
Comment (ianvernon) [reply pkg/policy/api/rule_validation_test.go:20]: Done.

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
@rlenglet I've redone the PR, PTAL

**Review by rlenglet — APPROVED** [review]
Great!

**Review by tgraf — APPROVED** [review]

**Pull request merged**